lhotse/dataset/speech_synthesis.py

```diff
@@ -15,38 +15,54 @@ class SpeechSynthesisDataset(torch.utils.data.Dataset):
     .. code-block::
 
         {
-            'audio': (B x NumSamples) tensor
-            'features': (B x NumFrames x NumFeatures) tensor
-            'tokens': list of lists of characters (str)
+            'audio': (B x NumSamples) float tensor
+            'features': (B x NumFrames x NumFeatures) float tensor
+            'tokens': (B x NumTokens) int tensor
+            'features_lens': (B x NumFrames) int tensor
+            'tokens_lens': (B x NumTokens) int tensor
         }
     """
 
-    def __init__(self, cuts: CutSet) -> None:
+    def __init__(
+        self,
+        cuts: CutSet,
+        feature_transforms: Union[Iterable[Callable], Callable] = [],
+        add_eos: bool = True,
+        add_bos: bool = True,
+    ) -> None:
         super().__init__()
+
         validate(cuts)
+        for cut in cuts:
+            assert (
+                len(cut.supervisions) == 1
+            ), "Only the Cuts with single supervision are supported."
+
         self.cuts = cuts
+        self.token_collater = TokenCollater(cuts, add_eos=add_eos, add_bos=add_bos)
 
-        # generate tokens from text
-        self.cut_id_to_token = {}
-        self.tokens = set()
-        for cut in cuts:
-            assert len(cut.supervisions) == 1, 'Only the Cuts with single supervision are supported.'
-            characters = list(cut.supervisions[0].text)
-            self.tokens.update(set(characters))
-            self.cut_id_to_token[cut.id] = characters
-        self.tokens = sorted(list(self.tokens))
+        if not isinstance(feature_transforms, Iterable):
```

Comment:
I think it has to be at least `Sequence`, `Iterable` can be a one-pass iterator

Reply:
good catch, fixing